extra/grpc/backend_outlines/backend_outlines.py

```diff
@@ -0,0 +1,82 @@
+"""
+This is the extra gRPC server for outlines of LocalAI
+"""
+from concurrent import futures
+import argparse
+import os
+import signal
+import sys
+import time
+
+import backend_pb2
+import backend_pb2_grpc
+
+import grpc
+
+import outlines.text.generate as generate
+import outlines.models as models
+
+_ONE_DAY_IN_SECONDS = 60 * 60 * 24
+
+# If MAX_WORKERS are specified in the environment use it, otherwise default to 1
+MAX_WORKERS = int(os.environ.get('PYTHON_GRPC_MAX_WORKERS', '1'))
+
+# Implement the BackendServicer class with the service methods
+class BackendServicer(backend_pb2_grpc.BackendServicer):
+    """
+    BackendServicer is the class that implements the gRPC service
+    """
+    def Health(self, request, context):
+        return backend_pb2.Reply(message=bytes("OK", 'utf-8'))
+
+    def LoadModel(self, request, context):
+        try:
+            # TODO: Need to make sure request.Model is a model name or path
+            if request.Model == "":
+                return backend_pb2.Result(success=False, message="Model name is empty")
+            # TODO: It is download model from huggingface.co, this should be optimized. The internal of this funciton is used transformers
+            # Maybe we can only load the model from local
+            self.model = models.transformers(request.Model)
+        except Exception as err:
+            return backend_pb2.Result(success=False, message=f"Unexpected {err=}, {type(err)=}")
+        return backend_pb2.Result(message="Model loaded successfully", success=True)
+
+    def Predict(self, request, context):
+        try:
+            #TODO: stop=["."] is a hard code, we need to make it configurable
+            output=generate.continuation(self.model, stop=["."])(self.prompt)
```

Comment:
it would make sense here to reuse the stopwords that are already sent as part of the request

Reply:
Done